ReactNativeClient/lib/folders-screen-utils.js

```diff
@@ -50,6 +50,10 @@ class FoldersScreenUtils {
 			this.refreshFolders();
 		}, 1000);
 	}
+
+	static cancelTimers() {
+		if (this.scheduleRefreshFoldersIID_) clearTimeout(this.scheduleRefreshFoldersIID_);
+	}
```

Comment:
Note that even if you cancel the timer, the refreshFolders function might still be running since it's async. Could that be a problem for the test units? One big issue I had with tests is they sometimes would work and sometimes fail randomly, and that's because there are still code running in the background. So just wondering if we might end up with this issue here.

Reply:
I think you're right.  I'll use your method here to be sure.

Looking further, `refreshFolders` and `scheduleRefreshFolders` are both called from the `generalMiddleware` function which is also async. So I think we need to cover this too, by using your method again, to ensure all the pending `BaseApplication` work is flushed and shut down before moving on to the other tests.

Reply:
PR updated with changes as suggested.

I also applied it to the other services scheduled by the `BaseApplication` so they are all cleaned up the same way during testing.